Projects/Embark/Sources/Viewables/Actions/TextActionSet.swift

```diff
@@ -35,18 +46,28 @@ extension TextActionSet: Viewable {
                 shouldAutoFocus: index == 0,
                 fieldStyle: .embarkInputSmall
             )
-            return (signal: view.addArranged(input), shouldReturn: input.shouldReturn, action: textAction)
+            
+            let isFirstName: Bool = textAction.data?.key == "firstName"
```

Comment:
You shouldn't use a hard-coded key value to identify this behaviour, what is the effect you want to achieve here? :)

Reply:
this is before I knew how embark worked!